### ISSUE TITLE
Error with Icons containing circles: some circles will be filled, although they shouldn't

### ISSUE THREAD

**Issue by scrapix** [tancredi/fantasticon#135]
Icon creation + SVG export tool: figma
OS: OS X BigSur
Browsers: Safari, Firefox, Chrome
fantasticon version: 1.1.1

**Problem** 
no matter how I create Icons, some circles get filled unexpectedly

**Icon Creation in 4 different ways**
- decision-tree2: only centered strokes
- decision-tree3: converted strokes to paths
- decision-tree4: unioned together everything
- decision-tree5: flattened result
- 
<img width="703" alt="image" src="https://user-images.githubusercontent.com/23144020/109778861-9a47a780-7c05-11eb-9d48-7801eeac1833.png">

**Result**
<img width="504" alt="image" src="https://user-images.githubusercontent.com/23144020/109778967-bfd4b100-7c05-11eb-9541-75657c066d61.png">

**Comment by tancredi** [comment]
Hi @scrapix - can you share your SVG? I would imagine there's a transparent fill circle somewhere in there :)

**Comment by scrapix** [comment]
Hi @tancredi , thanks for the quick reply. Sure, here are the SVGs:
[Archive.zip](https://github.com/tancredi/fantasticon/files/6075037/Archive.zip)

**Comment by scrapix** [comment]
I've double-checked the svg in Adobe illustrator and in Atom. There seems to be no additional object withing the SVG. Also interesting is, that all circles are copies of one object. So it's unclear why some get a fill, while others not. :-O

**Comment by tancredi** [comment]
Yes, `decision-tree5` looks fine and should work - in facts, I tested it in an icon set and it renders correctly post-generation for me.

<img width="123" alt="Screenshot 2021-03-03 at 10 08 04" src="https://user-images.githubusercontent.com/1300753/109789652-733ea500-7c08-11eb-8f3f-ac3a1102ec5c.png">

**Comment by tancredi** [comment]
Really strange - are you sure it's the same exact version of the SVG that was used for that reported test?

**Comment by scrapix** [comment]
Super strange indeed! Why would it work for you, but not for me

**So here is my terminal:**
<img width="978" alt="image" src="https://user-images.githubusercontent.com/23144020/109790299-8a35c500-7c11-11eb-9616-63648f086cab.png">

**Here is the whole directory with the source Icons + generated font:**
[icons-test.zip](https://github.com/tancredi/fantasticon/files/6075137/icons-test.zip)

**Note**
I've created (from scratch) the same Icon, with the same steps in Adobe Illustrator
--> Works perfectly

**Result**
<img width="477" alt="image" src="https://user-images.githubusercontent.com/23144020/109790501-c49f6200-7c11-11eb-91ad-bab7db7a7c52.png">

**Comment by stupidkitty** [comment]
Same on ubuntu 20.04

**Comment by samschooler** [comment]
I've had this issue with similar icons. The icons look correct on macOS Catalina, however are filled on macOS Big Sur.

**Comment by rhenriquez28** [comment]
I'm having the exact same issue

**Comment by giovanni-bertoncelli** [comment]
Same identical issue on Catalina... is there any workaround on this?

**Comment by misolori** [comment]
I've been encountering this issue for a number of years, even in the previous [icon-font-generator](https://github.com/Workshape/icon-font-generator) version, and I believe it does come down to how Figma is exporting SVGs. A simple square with outline strokes:

<img width="350" alt="CleanShot 2021-09-02 at 11 52 16@2x" src="https://user-images.githubusercontent.com/35271042/131900339-22ea6492-112e-42f4-b58b-dac203cd8858.png">

Generates this SVG code in Figma:

```svg
<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M14 2H2V14H14V2ZM1 1V15H15V1H1Z" fill="white"/>
</svg>
```
While the same icon in Illustrator is exported as:

```svg
<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M14 2V14H2V2H14ZM15 1H1V15H15V1Z" fill="white"/>
</svg>
```
The main difference is that Figma is using `fill-rule="evenodd"` and Illustrator does not and composes the path differently:

**Figma**
```svg
<path d="M14 2H2V14H14V2ZM1 1V15H15V1H1Z" />
```

**Illustrator**
```svg
<path d="M14 2V14H2V2H14ZM15 1H1V15H15V1Z" />
```

**Comment by uyarn** [comment]
same issue with `figma export` + `fill-rule="evenodd"`

**Comment by misolori** [comment]
I learned that there is a "[Fill Rule Editor](https://www.figma.com/community/plugin/771155994770327940/Fill-Rule-Editor)" Figma plugin that allows you to toggle the fill rules and properly export svgs to be used with font tools like fantasticon. I've been using this as an alternative and it's working well.

**Comment by receptiryaki** [comment]
[Fill Rule Editor](https://www.figma.com/community/plugin/771155994770327940/Fill-Rule-Editor) has not worked for me. But I've re-created the icon in [Vectornator](https://www.vectornator.io) software on MacOS. It exported without any issue now.

**Comment by vdcrea** [comment]
This package works like a charm https://www.npmjs.com/package/svg-reorient and fixed my output. After running it against the files all `clip-rule="evenodd"` are removed and the font is perfect.
